lib/tree.rb

```diff
@@ -16,51 +16,225 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n)
+  # Space Complexity: O(n)
   def add(key, value)
```

Comment:
Since the number of recursive calls never go above the depth of the tree, it's O(log n) for space complexity due to the recursive call stack.

This of course assumes the tree is balanced.